File: DependencyInjection/ServiceLocatorDIAntipattern/_2_ServiceLocator/Services/MsSqlBasedRepository.cs
using ServiceLocatorDIAntipattern._2_ServiceLocator.Interfaces;

namespace ServiceLocatorDIAntipattern._2_ServiceLocator.Services
{
  public interface IRepository
  {
    void Save(AcmeMessage message);
  }

  public class MsSqlBasedRepository : IRepository
  {
    private readonly DataDestination _sqlDataDestination = new SqlDataDestination();

    public void Save(AcmeMessage message)
    {
      message.WriteTo(_sqlDataDestination);
    }
  }
}

File: Java/dependency-injection/bastard-injection/src/main/java/com/github/grzesiek_galezowski/BastardInjection/Services/MsSqlBasedRepository.java
package com.github.grzesiek_galezowski.BastardInjection.Services;

import com.github.grzesiek_galezowski.BastardInjection.Interfaces.AcmeMessage;
import com.github.grzesiek_galezowski.BastardInjection.Interfaces.DataDestination;

public class MsSqlBasedRepository implements IRepository {
  private final DataDestination _dataDestination;

  public MsSqlBasedRepository() {
    this(new SqlDataDestination());
  }

  //for tests
  public MsSqlBasedRepository(DataDestination dataDestination) {
    _dataDestination = dataDestination;
  }

  public void save(AcmeMessage message) {
    message.writeTo(_dataDestination);
  }
}

